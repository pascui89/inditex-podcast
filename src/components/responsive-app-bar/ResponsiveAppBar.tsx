import {
  CircularProgress,
  Container,
  IconButton,
  Toolbar,
  Tooltip,
} from '@mui/material';
import AppBar from '@mui/material/AppBar';
import { StyledAvatar, StyledTypography } from '../styled';
import useResponsiveAppBar from './useResponsiveAppBar';

/**
 * ResponsiveAppBar Component
 *
 * Displays a responsive app bar with a logo, title, and loading spinner.
 */
export default function ResponsiveAppBar() {

  const { 
    state: { showSpinner }, 
    actions: { handleGoHomeRedirect } 
  } = useResponsiveAppBar();

  return (
    <AppBar
      color="default"
      position="fixed"
      className="shadow-lg"
    >
      <Container maxWidth="xl">
        <Toolbar
          disableGutters
          className="flex justify-between"
        >
          <div  className='flex items-center'>
            <Tooltip title="Go Home">
              <IconButton onClick={handleGoHomeRedirect}>
                <StyledAvatar
                  alt="Inditex PodCast"
                  src="logo.jpg"
                />
              </IconButton>
            </Tooltip>
            <StyledTypography
              variant="h6"
              noWrap
              onClick={handleGoHomeRedirect}
            >
              Podcaster
            </StyledTypography>
          </div>
          {showSpinner && <CircularProgress />}
        </Toolbar>
      </Container>
    </AppBar>
  );
}
