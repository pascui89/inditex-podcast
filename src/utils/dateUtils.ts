/**
 * Converts a date string to the specified date format.
 * @param dateString - The date string to be converted.
 * @returns The converted date string in the format "dd/mm/yyyy".
 */
export const toDateFormat = (dateString: string): string => {
  const date = new Date(dateString);
  const convertedDate = date.toLocaleDateString('es-ES', {
    day: '2-digit',
    month: '2-digit',
    year: 'numeric',
  });

  return convertedDate;
};

/**
 * Converts the given number of milliseconds to minutes and seconds format.
 * @param milliseconds - The number of milliseconds to be converted.
 * @returns The converted time in the format "mm:ss".
 */
export const millisecondsToMinutesSeconds = (milliseconds: number): string => {
  const totalSeconds = Math.floor(milliseconds / 1000);
  const minutes = Math.floor(totalSeconds / 60);
  const seconds = totalSeconds % 60;

  return `${minutes}:${seconds.toString().padStart(2, '0')}`;
};
