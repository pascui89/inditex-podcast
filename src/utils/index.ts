export { millisecondsToMinutesSeconds, toDateFormat } from './dateUtils';
