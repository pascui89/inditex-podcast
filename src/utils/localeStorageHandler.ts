import { Podcast } from "../models";

/**
 * Interface for the data stored in local storage.
 */
interface LocalStorageData {
    value: Podcast;
    expiry: number;
  }
  
  /**
   * Retrieves the data from local storage.
   * @returns The parsed local storage data.
   */
  export const getLocalStorageData = (): LocalStorageData | null => {
    return localStorage.getItem('podcastItems') ? JSON.parse(localStorage.getItem('podcastItems') || '') as LocalStorageData : null;
  };
  
  /**
   * Sets the data in local storage.
   * @param podcasts - The podcast data to be stored.
   */
  export const setLocalStorageData = (podcasts: Podcast): void => {
    const data = {
      value: podcasts,
      expiry: new Date().getTime() + (24 * 60 * 60 * 1000),
    } as LocalStorageData;
    localStorage.setItem('podcastItems', JSON.stringify(data));
  };
  