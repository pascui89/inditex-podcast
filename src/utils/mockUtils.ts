import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);
export const getMockStore = ( customStore = {} ) => {
    return mockStore({
        podcastReducer: {
            loading: true
        },
        episodeReducer: {
            loading: true,
            loadingEpisode: true
        },
        ...customStore
    });
}