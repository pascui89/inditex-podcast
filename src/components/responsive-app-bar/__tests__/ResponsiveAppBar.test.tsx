import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import ResponsiveAppBar from '../../../components/responsive-app-bar/ResponsiveAppBar';
import '@testing-library/jest-dom/extend-expect';

import { getMockStore } from '../../../utils/mockUtils';

global.React = React;

describe('ResponsiveAppBar', () => {

  it('should render', () => {
    const { container } = render(
      <Provider store={getMockStore()}>
        <BrowserRouter>
          <ResponsiveAppBar />
        </BrowserRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
