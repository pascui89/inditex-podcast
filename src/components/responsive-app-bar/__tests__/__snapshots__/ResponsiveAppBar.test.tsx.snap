// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveAppBar should render 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionFixed shadow-lg mui-fixed css-1ogdjvx-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular flex justify-between css-r6ewbb-MuiToolbar-root"
      >
        <div
          class="flex items-center"
        >
          <button
            aria-label="Go Home"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-e9g7qs-MuiAvatar-root"
            >
              <img
                alt="Inditex PodCast"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="logo.jpg"
              />
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-80v4v8-MuiTypography-root"
          >
            Podcaster
          </h6>
        </div>
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </header>
</div>
`;
